vignettes/sim_fixed_design_custom.Rmd

```diff
@@ -0,0 +1,406 @@
+  duration = rep(Inf, 3), 
+  rate = rep(0.001, 3)
+  )
+uncut_data_d <- sim_pw_surv(n = n,
+                            stratum = data.frame(stratum = "All"),
+                            block = c(rep("Control", 3), rep("Low-dose", 2), rep("High-dose", 2)),
+                            enroll_rate = enroll_rate,
+                            fail_rate = three_arm_fail_rate,
+                            dropout_rate = three_arm_dropout_rate
+                            )
+```
+
+# Step 2: Cut data
+
+The `get_analysis_date()` derives analysis date for interim/final analysis given multiple conditions, see [the help page of `get_analysis_date()` at the pkgdown website](https://merck.github.io/simtrial/reference/get_analysis_date.html). 
+
+Users can cut for analysis at the 24th month and there are 300 events, whichever arrives later. This is equivalent to `timing_type = 4` in `sim_fixed_n()`.
+```{r}
+cut_date_a <- get_analysis_date(data = uncut_data,
+                                planned_calendar_time = 24,
+                                target_event_overall = 300)
+```
+
+Users can also cut by the maximum of targeted 300 event and minimum follow-up 12 months. This is equivalent to `timing_type = 5` in `sim_fixed_n()`.
+```{r}
+cut_date_b <- get_analysis_date(data = uncut_data,
+                                min_followup = 12,
+                                target_event_overall = 300)
+```
+
+Users can cut data when there are 300 events, with maximum time extension to reach targeted events of 24 months. This is not enabled in `timing_type` of `sim_fixed_n()`.
+
+```{r}
+cut_date_c <- get_analysis_date(data = uncut_data,
+                                max_extension_for_target_event = 12,
+                                target_event_overall = 300)
+```
+
+Users can cut data after 12 months followup when 80\% of the patients are enrolled in the overall population as below. This is not enabled in `timing_type` of `sim_fixed_n()`.
+
+```{r}
+cut_date_d <- get_analysis_date(data = uncut_data,
+                                min_n_overall = 100 * 0.8,
+                                min_followup = 12)
+```
+
+More examples are available in the [reference page of `get_analysis_date()`](https://merck.github.io/simtrial/reference/get_analysis_date.html) For illustration purposes, we will focus on scenario d) for the following discussion.
+
+```{r}
+cut_date <- cut_date_d
+cat("The cutoff date is ", round(cut_date, 2))
+
+cut_data <- uncut_data |> cut_data_by_date(cut_date)
+cut_data |> head() |> gt() |> tab_header(paste0("An Overview of TTE data Cut at ", round(cut_date, 2), "Months"))
+```
+
+# Step 3: Run tests
+
+The `simtrial` package provides many options for testing methods, including (weighted) logrank tests, RMST test, milestone test, and MaxComboi test, see the [Section "Compute p-values/test statistics" at the pkgdown reference page] (https://merck.github.io/simtrial/reference/index.html#compute-p-values-test-statistics).
```

Comment:
Looking at the other vignettes, I believe our convention in this repository is to refer to R package names without any extra syntax

Reply:
Is there a way we could refer to a vignette of an R pkg?

Reply:
You mean so that you can click on a link and view it? The easiest would probably be to simply create a link to the vignette in the pkgdown site if the package has one

Reply:
> You mean so that you can click on a link and view it? 

Yes.

> The easiest would probably be to simply create a link to the vignette in the pkgdown site if the package has one.

This is what I put on this PR. The vignette link is saved in the Rmd file. Could you please re-review it?

Reply:
The links to the pkgdown website are fine.

My original comment was related to how you are typing out the simtrial package. In these vignettes, you are surrounding simtrial with backticks. We don't do that in the other vignettes in this package. I am suggesting you remove the backticks for consistency, but I don't feel strongly about it. You can leave them if you like.

Reply:
I got you! Sorry, I misunderstood your comments. I make a new commit to get the typing of simtrial consistent with the other vignettes.